src/resolve-repository-reference.ts

```diff
@@ -0,0 +1,107 @@
+import { directoryExists } from '@metamask/utils/node';
+import path from 'path';
+
+import { ensureMetaMaskRepositoriesLoaded } from './ensure-metamask-repositories-loaded';
+import { createModuleLogger, projectLogger } from './logging-utils';
+
+const log = createModuleLogger(projectLogger, 'resolve-repository-reference');
+
+type ResolvedRepositoryReference = {
+  repositoryShortname: string;
+  repositoryDirectoryPath: string;
+  repositoryDirectoryExists: boolean;
+  isKnownMetaMaskRepository: boolean;
+};
+
+/**
+ * A "repository reference" may be:
+ *
+ * 1. The path of an existing directory relative to this tool's working
+ * directory.
+ * 2. The absolute path of an existing directory.
+ * 3. The path to a previously cloned MetaMask repository.
+ * 4. The name of a MetaMask repository.
+ *
+ * This function determines which one it is.
```

Comment:
This description seems a bit misleading. I was expecting it to return an enum to describe which one of these reference types it is, but in actuality that's not what this function is for at all.

It resolves the reference to this metadata object about the directory, but I'm not finished reading this yet so I'm not sure how to best describe it in the "cached directory" case

Reply:
I see, good feedback. I've pushed up some updates here, hopefully this makes more sense: [7a88546](https://github.com/MetaMask/module-lint/pull/30/commits/7a8854677fc55716a3e9a3cde695840e610b803c)